streamer/src/nonblocking/quic.rs

```diff
@@ -1152,7 +1152,13 @@ pub mod test {
         for _ in 0..num_bytes {
             s1.write_all(&[0u8]).await.unwrap();
         }
-        s1.finish().await.unwrap();
+        let result = s1.finish().await;
+        match result {
+            Err(quinn::WriteError::UnknownStream) => {
```

Comment:
> We expect a result of Ok or Stopped from peer.

It might be a bit more future-proof to only accept valid results, rather than only checking for invalid result.
This way, if another error case is added to [`quinn::WriteError`](https://docs.rs/quinn/latest/quinn/enum.WriteError.html) it would be a failure, by default.

Another approach is to list all possible cases explicitly, both positive and negative.
In this case, any new future addition will be marked by the compiler
Someone will have to decide what is the right meaning for that new error, when upgrading the library.

By the way, `quinn::WriteError` has two more cases: `ConnectionLost` and `ZeroRttRejected`.
Do you think the test should still pass if the error is `ConnectionLost`?
My guess is that `ZeroRttRejected` is unlikely, but should the test pass when `ZeroRttRejected` is returned?